### ISSUE TITLE
environments_identities_featurestates_update requires environment with key

### ISSUE THREAD

**Issue by rcmosher** [SolidStateGroup/bullet-train-api#2275]
**Describe the bug**

Calling PUT endpoint `/api/v1/environments/{environment_api_key}/identities/{identity_pk}/featurestates/{id}/` with the enivronment_api_key supplied, and even with the server-side api key for the environment in the header fails unless the `environment` property is supplied.

Other similar endpoints don't need the environment explicitly specified, pulling it from the environment_api_key instead. For example using POST create the feature state

**To Reproduce**

1. Use the specified endpoint.
2. Don't provide the environment property, only the api key
3. Receive a 400 error with string "Invalid Identifier"

**Expected behavior**

Call would be successful since the environment can be determined from the key.
The error message is also unclear since it is not a problem with the identifier property.

**How are you running Flagsmith?**

- [x] Self Hosted with Docker
- [ ] Self Hosted with Kubernetes
- [ ] SaaS at flagsmith.com
- [ ] Some other way

**Additional context**

**Comment by matthewelwell** [comment]
Thanks @rcmosher, we will add this to the backlog and take a look when we find time, however, since we are hoping in the not too distant future to rebuild the API, taking into account more client friendly use cases, this likely is not going to be done any time soon. If you'd like to submit a PR, we'd be more than willing to take a look of course.

**Comment by rcmosher** [comment]
@matthewelwell  That's good to know. It could definitely be more client friendly. My natural expectation was if I know the global feature ID (or name), environment, and optionally identity, that should be enough to read or manipulate environment and identity overrides.

There could also be some better documentation around properties, and differences between All and List endpoints.

**Comment by matthewelwell** [comment]
Yep, these are some of the key things that will be driving the rework of the API.
